src/scales/ordinal.js

```diff
@@ -116,7 +116,7 @@ function inferDomain(channels) {
 // If all channels provide a consistent hint, propagate it to the scale.
 function inferSymbolHint(channels) {
   const hint = {};
-  for (const {hint: channelHint} of channels) {
+  for (const {hint: channelHint = {}} of channels) {
```

Comment:
It looks like this code requires that each channel specify a hint. If the hint is optional, then I think we’ll need to change this loop to instead ignore channels that don’t have a hint. Currently it looks like this code will now consider an undefined hint a hint, which might not be what we want.

Reply:
Also, if this is fixing a bug, it’d be helpful to have a comment or an issue to reference. 🙏

Reply:
yes, I should have done this before!  issue #806

Reply:
ty!